Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -126,15 +142,61 @@ def draw(self, pen):
         how that works.
         """
         glyph, offset = self._getGlyphAndOffset()
+
+        if self.glyphSet.locationStack:
+            offset = 0  # Offset should only apply at top-level
+
+        if glyph.isVarComposite():
+            self._drawVarComposite(glyph, pen, False)
+            return
+
         glyph.draw(pen, self.glyphSet.glyfTable, offset)
 
     def drawPoints(self, pen):
         """Draw the glyph onto ``pen``. See fontTools.pens.pointPen for details
         how that works.
         """
         glyph, offset = self._getGlyphAndOffset()
+
+        if self.glyphSet.locationStack:
+            offset = 0  # Offset should only apply at top-level
+
+        if glyph.isVarComposite():
+            self._drawVarComposite(glyph, pen, True)
+            return
+
         glyph.drawPoints(pen, self.glyphSet.glyfTable, offset)
 
+    def _drawVarComposite(self, glyph, pen, isPointPen):
+
+        from fontTools.ttLib.tables._g_l_y_f import VarComponentFlags
+
+        for comp in glyph.components:
+
+            # TODO Handle missing attributes?
+            t = Transform()
+            t = t.translate(
+                comp.translateX + comp.tCenterX, comp.translateY + comp.tCenterY
+            )
+            t = t.rotate(comp.rotation / 180 * math.pi)
+            t = t.scale(comp.scaleX, comp.scaleY)
+            t = t.skew(-comp.skewX / 180 * math.pi, comp.skewY / 180 * math.pi)
+            t = t.translate(-comp.tCenterX, -comp.tCenterY)
+
+            if isPointPen:
+                tPen = TransformPointPen(pen, t)
+            else:
+                tPen = TransformPen(pen, t)
+
+            self.glyphSet.pushLocation(
```

Comment:
If `pushLocation` were a context manager, you could use `with`, and get rid of the pop call.